lib/util/http/host-rules.ts

```diff
@@ -162,6 +163,18 @@ export function applyHostRules<GotOptions extends HostRulesGotOptions>(
   if (!hasProxy() && foundRules.enableHttp2 === true) {
     options.http2 = true;
   }
+
+  if (
+    foundRules.certificateAuthority !== null ||
+    foundRules.privateKey !== null ||
+    foundRules.certificate !== null
+  ) {
+    options.https = {
+      certificateAuthority: foundRules.certificateAuthority,
+      key: foundRules.privateKey,
+      certificate: foundRules.certificate,
```

Comment:
Will this work if only 1/3 or 2/3 of these are defined?

Reply:
the `key` and `certificate` are the key bit in this. for our case CA needs to be specified.

Reply:
Should we validate that programmatically? At the very least it should be documented (Eg which options are required together)

Reply:
Maybe not. these options seems to be passed into `got` and `got` also seems to use them straight away to create native https request options. see

https://github.com/sindresorhus/got/blob/944caa8ce8151d7b18ab17c4130b18d622c04ebe/source/core/options.ts#L2432-L2434

https://github.com/sindresorhus/got/blob/944caa8ce8151d7b18ab17c4130b18d622c04ebe/source/core/index.ts#L1121

Reply:
Also there's an other use case in our environment is that. 

The client only need to load the internal certificateAuthority to verify the server's SSL cert's validity. 
The server however doesn't require the client to have a valid internal cert to perform mTLS authentication.

I think these different use cases are the reason why `got` doesn't do any verification on those input params

Reply:
If you're saying that _any_ combination of these three can be valid (e.g. each of them alone, or any two of them together without the third), then we are fine. But if there are invalid combinations which simply don't make sense, we should validate that as part of config validation to alert the user.